### ISSUE TITLE
CLI 1.5.0-beta.1 and Angular 5.0.0-beta.7 error TS5052 due to tsconfig sourceMap config

### ISSUE THREAD

**Issue by sgwatgit** [angular/angular-cli#7826]
### Bug Report or Feature Request (mark with an `x`)
```
- [ X] bug report -> please search issues before submitting
- [ ] feature request
```

### Versions.
@angular/cli: 1.5.0-beta.1
node: 8.3.0
os: win32 x64
@angular/animations: 5.0.0-beta.7
@angular/cdk: 2.0.0-beta.10
@angular/common: 5.0.0-beta.7
@angular/compiler: 5.0.0-beta.7
@angular/core: 5.0.0-beta.7
@angular/forms: 5.0.0-beta.7
@angular/http: 5.0.0-beta.7
@angular/material: 2.0.0-beta.10
@angular/platform-browser: 5.0.0-beta.7
@angular/platform-browser-dynamic: 5.0.0-beta.7
@angular/router: 5.0.0-beta.7
@angular/cli: 1.5.0-beta.1
@angular/compiler-cli: 5.0.0-beta.7
typescript: 2.4.2

### Repro steps.
ng build --prod

### The log given by the failure.
`ERROR in error TS5052: Option 'mapRoot' cannot be specified without specifying option 'sourceMap'.`

### Desired functionality.
Successful build

### Mention any other details that might be useful.
root tsconfig.json

```
{
  "compilerOptions": {
    "baseUrl": "src",
    "declaration": false,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "noUnusedLocals": true,
    "noUnusedParameters": false,
    "allowUnreachableCode": false,
    "lib": [
      "es2017",
      "dom"
    ],
    "mapRoot": "./",
    "module": "es2015",
    "moduleResolution": "node",
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "target": "es5",
    "typeRoots": [
      "../node_modules/@types",
      "./typings.d.ts"
    ],
    "types": [
      "node"
    ]
  }
}
```
src/tsconfig.app.json
```
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "outDir": "../out-tsc/app",
    "module": "es2015",
    "baseUrl": "",
    "types": []
  },
  "exclude": [
    "test.ts",
    "**/*.spec.ts"
  ]
}
```

**Comment by filipesilva** [comment]
`ng build --prod` has sourcemaps disabled by default, so with your tsconfig.json builds will fail. You have to remove the `mapRoot` property from your tsconfig and the CLI will manage it for you.

**Comment by filipesilva** [comment]
My bad, I thought mapRoot was being managed by the CLI but it actually wasn't. @clydin fixed it in #7862
